### DISCUSSION TITLE
How to predict a generic forecasting interval

### DISCUSSION THREAD

**Discussion by corradomio** [alan-turing-institute/sktime#5281] [Q&A]
We suppose to have a monthly dataset D=[DX|Dy] in the interval [1950/01-2000/12] and the model M permits to specify the past window, for example 12 months, to use during the learning step. This means that the prediction requires to retrieve the past data in the specified window:

`
m = M(windows=12)
m.fit(Dy, DX)
`

Now, we are able to predict, for example, the next 6 months using the forecasting horizon fh=[2001/01-2001/06] and Dfh

`
p = m.predict(fh, Dfh)
`

But we suppose that we have to predict 6 months using the forecasting horizon fh'=[2023/06-2023/12]'  (and the Dfh' in the same interval).
Obviously, to use the model in this way it has no sense:

`
p = m.predict(fh', Dfh')
`

because there are missing the previous 12 months.
In "theory" it is possible to use the model passing the PAST dataset Dp=[DXp,Dyp] in the interval [2022/06-2023/05], that is, the hypotetical syntax could be:

`
p = m.predict(fh', Dfh', yh=Dyp, Xh=DXp)
`

where 'yh' and 'Xh' are the 'historical' y and X.

The question is: is it possible to use the models in this way? And how?
Because the documentation is not clear or I don't have found how to do.
I suppose it is necessary to use 'm.update' and m.update_predict', but it is not clear how.

**Comment by fkiraly** [comment]
@corradomio, you can do what you want - to ingest additional data, you need to use the `update` method.

Using the syntax in your example, you would do:

```python
m = M(windows=12) m.fit(Dy, DX)
p = m.predict(fh, Dfh)
m.update(Dyp, Dxp)
m.predict(Dfh')
```

The only restriction is that `fh` cannot be changed once passed, and should be relative if you use `update`.

**Comment by corradomio** [comment]
Thanks a lot. I had found it.
